### PR TITLE
Adds `COOK_TEST_DEFAULT_SUBMIT_POOL` env var

### PR DESCRIPTION
**Pull request by dposada**
## Changes proposed in this PR

- allowing `COOK_TEST_DEFAULT_SUBMIT_POOL` to dictate the pool to use on submission, if the calling path does not set one
- ensuring that test users have the default share and quota 

## Why are we making these changes?

We don't always want to use Cook Scheduler's configured default pool for our test runs.

The share and quota fixes are needed because some tests can leave share/quota at small values if they terminate unexpectedly.

### CODE REVIEW

File: integration/tests/cook/test_impersonation.py [hunk 1]
```diff
@@ -119,6 +119,9 @@ def test_self_impersonate(self):
         user1 = self.user_factory.new_user()
         job_uuids = []
         try:
+            with self.user_factory.admin():
```
Comment (scrosby) [inline integration/tests/cook/test_impersonation.py]: This change seems unrelated to the PR. Split off?

File: integration/tests/cook/test_multi_user.py [hunk 2]
```diff
@@ -45,6 +45,9 @@ def test_job_delete_permission(self):
 
     def test_group_delete_permission(self):
         user1, user2 = self.user_factory.new_users(2)
+        with self.user_factory.admin():
```
Comment (scrosby) [inline integration/tests/cook/test_multi_user.py]: This change seems unrelated to the PR. Split off?

File: integration/tests/cook/util.py [hunk 3]
```diff
@@ -490,6 +490,10 @@ def minimal_group(**kwargs):
     return dict(uuid=str(make_temporal_uuid()), **kwargs)
 
 
+def default_submit_pool():
```
Comment (scrosby) [inline integration/tests/cook/util.py]: Should have a comment here. Including saying why its OK for response to be None. Also consider inlining this into the caller to make the Noneness of a potential response more visible.

File: integration/tests/cook/test_impersonation.py (hunk 1 above)
Comment (scrosby) [reply integration/tests/cook/test_impersonation.py]: Why is this code in this specific location, in this test? This looks to be setup/reset code that belongs somewhere else.

File: integration/tests/cook/test_multi_user.py (hunk 2 above)
Comment (scrosby) [reply integration/tests/cook/test_multi_user.py]: Why is this code in this specific location, in this test? This looks to be setup/reset code that belongs somewhere else.

File: integration/tests/cook/util.py (hunk 3 above)
Comment (dposada) [reply integration/tests/cook/util.py]: Good call, added a docstring.

File: integration/tests/cook/test_multi_user.py (hunk 2 above)
Comment (dposada) [reply integration/tests/cook/test_multi_user.py]: Yep, this is unrelated to the default submit pool change. That being said, I saw failures for these tests on this branch, so it seems appropriate to fix them on this branch.

As for why it's here, the resetting needs to happen for the specific user the test is using, which is determined in the test.

File: integration/tests/cook/test_impersonation.py (hunk 1 above)
Comment (dposada) [reply integration/tests/cook/test_impersonation.py]: Yep, this is unrelated to the default submit pool change. That being said, I saw failures for these tests on this branch, so it seems appropriate to fix them on this branch.

As for why it's here, the resetting needs to happen for the specific user the test is using, which is determined in the test.

Comment (scrosby) [reply integration/tests/cook/test_impersonation.py]: Thanks for explanation. 

Are the test failures caused by the intended code changes in this branch?

File: integration/tests/cook/test_multi_user.py (hunk 2 above)
Comment (scrosby) [reply integration/tests/cook/test_multi_user.py]: Thanks for explanation.

Are the test failures caused by the intended code changes in this branch?

File: integration/tests/cook/test_impersonation.py (hunk 1 above)
Comment (dposada) [reply integration/tests/cook/test_impersonation.py]: Nope, they were caused by a test run that was aborted and left a test user's quota in a too-low state.

File: integration/tests/cook/test_multi_user.py (hunk 2 above)
Comment (dposada) [reply integration/tests/cook/test_multi_user.py]: Nope, they were caused by a test run that was aborted and left a test user's quota in a too-low state.

File: integration/tests/cook/test_impersonation.py (hunk 1 above)
Comment (scrosby) [reply integration/tests/cook/test_impersonation.py]: *hmms* this still smells a bit off. Shouldn't we have something that resets the whole cluster to a known good state, if this can go wrong?

Comment (dposada) [reply integration/tests/cook/test_impersonation.py]: Good call, moved these calls to the `UserFactory`

File: integration/tests/cook/test_multi_user.py (hunk 2 above)
Comment (dposada) [reply integration/tests/cook/test_multi_user.py]: Moved these calls to the `UserFactory`

**Comment by dposada** [comment]
@scrosby Resolved, thanks
